lib/omniauth/strategy.rb

```diff
@@ -335,7 +335,7 @@ def mock_request_call
 
     def mock_callback_call
       setup_phase
-      @env['omniauth.origin'] = session.delete('omniauth.origin')
+      @env['omniauth.origin'] ||= session.delete('omniauth.origin')
```

Comment:
This has the side effect of leaving the omniauth.origin in the session if it's already been set

Reply:
Good point, updated